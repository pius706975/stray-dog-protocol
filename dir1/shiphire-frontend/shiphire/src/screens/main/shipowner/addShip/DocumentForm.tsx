import { FormikHelpers } from 'formik';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { DocumentPickerResponse } from 'react-native-document-picker';
import { ProgressBar, View } from 'react-native-ui-lib';
import { useDispatch } from 'react-redux';
import { DynamicForm, ScreenLayout } from '../../../../components';
import { ArrowIcon, Color } from '../../../../configs';
import { useGetAddShipDynamicForm } from '../../../../hooks';
import { addShipSlice, progressIndicatorSlice } from '../../../../slices';
import { DocumentFormProps, DynamicFormType } from '../../../../types';
import { useIsFocused } from '@react-navigation/native';

const DocumentForm: React.FC<DocumentFormProps> = ({ navigation }) => {
    const dispatch = useDispatch();
    const { t } = useTranslation('home');
    const mutationGetForm = useGetAddShipDynamicForm();
    const [dynamicForm, setDynamicForm] = React.useState<DynamicFormType[]>([]);
    const { addDocument, removeDocument } = addShipSlice.actions;
    const { showProgressIndicator, hideProgressIndicator } =
        progressIndicatorSlice.actions;
    const isFocused = useIsFocused();

    const filteredForm = dynamicForm.filter(item => {
        return (
            item.dynamicInput.active &&
            item.dynamicInput.templateType === 'shipDoc'
        );
    });
    const sortedForm = filteredForm.sort(
        (a, b) => a.dynamicInput.order - b.dynamicInput.order,
    );

    const handleSubmit = async (value: any, actions: FormikHelpers<any>) => {
        const hasErrors = Object.entries(value).some(([label, item]) => {
            return (
                Object.keys(item as any).length === 0 ||
                Object.keys(item as any).length === 1
            );
        });

        if (hasErrors) {
            Object.entries(value).forEach(([label, item]) => {
                if (
                    Object.keys(item as any).length === 0 ||
                    Object.keys(item as any).length === 1
                ) {
                    actions.setFieldError(label, t('ShipOwner.textSelectFile'));
                }
            });
        } else {
            await Promise.all(
                Object.entries(value).map(([label, item]) => {
                    console.log(item);

                    return dispatch(
                        addDocument({
                            ...(item as DocumentPickerResponse & {
                                docExpired?: string | undefined;
                            }),
                            label,
                        } as DocumentPickerResponse & { label: string }),
                    );
                }),
            );

            navigation.navigate('ImageForm');
        }

        actions.setSubmitting(false);
    };

    React.useEffect(() => {
        const unsubscribe = navigation.addListener('beforeRemove', event => {
            event.preventDefault();
            dispatch(removeDocument());
            navigation.dispatch(event.data.action);
        });

        return unsubscribe;
    }, [navigation]);

    React.useEffect(() => {
        if (isFocused) {
            dispatch(removeDocument());
            dispatch(showProgressIndicator());
            mutationGetForm.mutate(undefined, {
                onSuccess: data => {
                    setDynamicForm(data.data.data.dynamicForms);
                    dispatch(hideProgressIndicator());
                },
            });
        }
    }, [isFocused]);

    return (
        <ScreenLayout backgroundColor={'light'} testId={'documentFormAddShip'}>
            <View marginH-20 marginV-26>
                <ProgressBar
                    progress={75}
                    style={{
                        backgroundColor: Color.secColor,
                        height: 10,
                        marginBottom: 20,
                    }}
                    progressColor={Color.primaryColor}
                />
                <DynamicForm
                    btnTitle={t('ShipOwner.buttonNext')}
                    data={sortedForm}
                    onSubmit={handleSubmit}
                    rightIcon={<ArrowIcon />}
                />
            </View>
        </ScreenLayout>
    );
};

export default DocumentForm;
