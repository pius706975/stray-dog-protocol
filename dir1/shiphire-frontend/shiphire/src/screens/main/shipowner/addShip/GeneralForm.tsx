import { FormikHelpers } from 'formik';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { FlatList } from 'react-native-gesture-handler';
import { ProgressBar, View } from 'react-native-ui-lib';
import { useDispatch } from 'react-redux';
import { CustomText, DynamicForm, ScreenLayout } from '../../../../components';
import { ArrowIcon, Color } from '../../../../configs';
import { useGetAddShipDynamicForm } from '../../../../hooks';
import { addShipSlice } from '../../../../slices';
import {
    DynamicFormType,
    GeneralFormAddShipRequest,
    GeneralFormProps,
} from '../../../../types';
import { useIsFocused } from '@react-navigation/native';

const GeneralForm: React.FC<GeneralFormProps> = ({ navigation }) => {
    const dispatch = useDispatch();
    const { t } = useTranslation('home');
    const mutationGetForm = useGetAddShipDynamicForm();
    const [dynamicForm, setDynamicForm] = React.useState<DynamicFormType[]>([]);
    const { addGeneralData } = addShipSlice.actions;
    const isFocused = useIsFocused();

    React.useEffect(() => {
        if (isFocused) {
            mutationGetForm.mutate(undefined, {
                onSuccess: data => {
                    setDynamicForm(data.data.data.dynamicForms);
                },
            });
        }
    }, [isFocused]);
    const filteredForm = dynamicForm.filter(
        item =>
            item.dynamicInput.active &&
            item.dynamicInput.templateType === 'generalAddShip',
    );
    const sortedForm = filteredForm.sort(
        (a, b) => a.dynamicInput.order - b.dynamicInput.order,
    );

    const onHandleSubmit = (
        values: any,
        actions: FormikHelpers<GeneralFormAddShipRequest>,
    ) => {
        actions.setSubmitting(true);
        dispatch(addGeneralData(values));
        actions.setSubmitting(false);
        navigation.navigate('SpecificForm');
    };

    return (
        <FlatList
            data={['']}
            keyExtractor={() => 'dummyKey'}
            renderItem={() => (
                <ScreenLayout
                    backgroundColor={'light'}
                    testId="generalFormAddShip">
                    <View marginH-20 marginV-26 flex style={{ gap: 20 }}>
                        <ProgressBar
                            progress={25}
                            style={{
                                backgroundColor: Color.secColor,
                                height: 10,
                            }}
                            progressColor={Color.primaryColor}
                        />
                        <View style={{ gap: 10 }}>
                            <CustomText
                                color="darkTextColor"
                                fontSize="xl2"
                                fontFamily="semiBold">
                                {t('ShipOwner.textAddGeneralForm')}
                            </CustomText>
                            <DynamicForm
                                data={sortedForm}
                                btnTitle={t('ShipOwner.buttonNext')}
                                rightIcon={<ArrowIcon />}
                                onSubmit={onHandleSubmit}
                            />
                        </View>
                    </View>
                </ScreenLayout>
            )}
        />
    );
};

export default GeneralForm;
