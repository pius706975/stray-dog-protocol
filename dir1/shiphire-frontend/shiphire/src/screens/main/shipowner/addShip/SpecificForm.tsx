import { FormikHelpers } from 'formik';
import React from 'react';
import { ProgressBar, View } from 'react-native-ui-lib';
import { useDispatch, useSelector } from 'react-redux';
import { CustomText, DynamicForm, ScreenLayout } from '../../../../components';
import { ArrowIcon, Color } from '../../../../configs';
import {
    DynamicFormType,
    RootState,
    SpecificFormAddShipRequest,
    SpecificFormProps,
} from '../../../../types';

import { useGetAddShipDynamicForm } from '../../../../hooks';

import { useTranslation } from 'react-i18next';
import { addShipSlice, progressIndicatorSlice } from '../../../../slices';
import { useIsFocused } from '@react-navigation/native';

const SpecificForm: React.FC<SpecificFormProps> = ({ navigation }) => {
    const dispatch = useDispatch();
    const { t } = useTranslation('home');
    const mutationGetForm = useGetAddShipDynamicForm();
    const generalFormData = useSelector((state: RootState) => state.addShip);
    const shipSpecData = useSelector((state: RootState) => state.shipSpecSlice);
    const [dynamicForm, setDynamicForm] = React.useState<DynamicFormType[]>([]);
    const { addSpesificData } = addShipSlice.actions;
    const { showProgressIndicator, hideProgressIndicator } =
        progressIndicatorSlice.actions;
    const isFocused = useIsFocused();

    const filteredForm = dynamicForm.filter(item => {
        const camelCaseSpecName = generalFormData.category
            .split(' ')
            .map((word, index) =>
                index === 0
                    ? word.toLowerCase()
                    : word.charAt(0).toUpperCase() +
                      word.slice(1).toLowerCase(),
            )
            .join('');
        return (
            (item.dynamicInput.templateType === 'spesificAddShip' ||
                item.dynamicInput.templateType ===
                    `${camelCaseSpecName}Spesific`) &&
            item.dynamicInput.active
        );
    });

    const sortedForm = filteredForm.sort(
        (a, b) => a.dynamicInput.order - b.dynamicInput.order,
    );

    const handleSubmit = (value: any, actions: FormikHelpers<any>) => {
        const specifications = value.shipSpec.map((specName: string) => {
            const camelCaseSpecName = specName
                .split(' ')
                .map((word, index) =>
                    index === 0
                        ? word.toLowerCase()
                        : word.charAt(0).toUpperCase() +
                          word.slice(1).toLowerCase(),
                )
                .join('');

            const specValue = value[camelCaseSpecName];

            return {
                name: specName,
                value: specValue || '',
            };
        });

        const spesificDataValue: SpecificFormAddShipRequest = {
            height: value.height,
            length: value.length,
            width: value.width,
            facilities: value.shipFacility,
            specifications: specifications,
        };

        dispatch(addSpesificData(spesificDataValue));
        navigation.navigate('DocumentForm');

        actions.setSubmitting(false);
    };

    React.useEffect(() => {
        if (isFocused) {
            dispatch(showProgressIndicator());
            mutationGetForm.mutate(undefined, {
                onSuccess: data => {
                    setDynamicForm(data.data.data.dynamicForms);
                    dispatch(hideProgressIndicator());
                },
            });
        }
    }, [isFocused]);

    React.useEffect(() => {
        let adjustedBargeSpecSpec: Record<string, DynamicFormType> = {};

        shipSpecData.shipSpecSlice.forEach(label => {
            // Cari item dalam filteredForm yang sesuai dengan label
            const matchingItem = dynamicForm.find(
                item => item.dynamicInput.label === label,
            );

            // Jika ada item yang sesuai, salin data ke objek adjustedBargeSpecSpec
            if (matchingItem) {
                adjustedBargeSpecSpec[label] = matchingItem;
            }
        });

        Object.entries(adjustedBargeSpecSpec).forEach(([label, item]) => {
            const index = sortedForm.findIndex(
                formItem => formItem.dynamicInput.label === label,
            );

            if (index !== -1) {
                sortedForm[index] = item as DynamicFormType;
            } else {
                // Jika item dengan label tersebut tidak ada dalam filteredForm, tambahkan item baru
                sortedForm.push(item as DynamicFormType);
            }
        });
    }, [shipSpecData]);

    return (
        <ScreenLayout backgroundColor={'light'} testId={'spesificFormAddShip'}>
            <View marginH-20 marginV-26 style={{ gap: 20 }}>
                <ProgressBar
                    progress={50}
                    style={{ backgroundColor: Color.secColor, height: 10 }}
                    progressColor={Color.primaryColor}
                />
                <View style={{ gap: 10 }}>
                    <CustomText
                        color="darkTextColor"
                        fontSize="xl2"
                        fontFamily="semiBold">
                        2. {generalFormData.category}{' '}
                        {t('ShipOwner.textAddSpecificForm')}
                    </CustomText>
                    <DynamicForm
                        btnTitle={t('ShipOwner.buttonNext')}
                        data={sortedForm}
                        onSubmit={handleSubmit}
                        rightIcon={<ArrowIcon />}
                    />
                </View>
            </View>
        </ScreenLayout>
    );
};

export default SpecificForm;
