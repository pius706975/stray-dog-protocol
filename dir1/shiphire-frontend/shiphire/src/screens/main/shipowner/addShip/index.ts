export {default as GeneralForm} from './GeneralForm';
export {default as SpecificForm} from './SpecificForm';
export {default as DocumentForm} from './DocumentForm';
export {default as ImageForm} from './ImageForm';
