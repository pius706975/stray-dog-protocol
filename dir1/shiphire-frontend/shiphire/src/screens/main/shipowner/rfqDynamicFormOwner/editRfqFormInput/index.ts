export { default as EditRFQInputFormOwner } from './EditRFQInputFormOwner';
