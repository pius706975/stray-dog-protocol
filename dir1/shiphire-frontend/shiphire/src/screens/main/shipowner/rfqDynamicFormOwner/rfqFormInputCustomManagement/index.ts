export { default as RFQFormInputCustomManagementOwner } from './RFQFormInputCustomManagementOwner';
