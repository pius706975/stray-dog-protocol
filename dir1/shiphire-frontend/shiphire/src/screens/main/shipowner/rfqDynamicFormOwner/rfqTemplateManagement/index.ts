export { default as RFQTemplateOwnerManagement } from './RFQTemplateOwnerManagement';
