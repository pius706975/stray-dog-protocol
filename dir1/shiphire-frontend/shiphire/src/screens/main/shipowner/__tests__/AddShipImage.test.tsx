import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { MainOwnerStackParamList } from '../../../../types';
import { QueryClient, QueryClientProvider } from 'react-query';
import httpRequest from '../../../../services/api';
import MockAdapter from 'axios-mock-adapter';
import { Provider, useSelector } from 'react-redux';
import store from '../../../../store';
import { NavigationContainer } from '@react-navigation/native';
import { ImageForm } from '../addShip';
import {
    useGetAddShipDynamicForm,
    useSubmitShip,
    useSubmitShipDocument,
    useSubmitShipImage,
} from '../../../../hooks';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import { setMockImagePickerResolve } from '../../../../jest/setup';
import { RFQTemplateOwnerManagement } from '../rfqDynamicFormOwner/rfqTemplateManagement';

jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn(),
}));

jest.mock('react-native-image-crop-picker', () => ({
    openPicker: jest.fn(() =>
        Promise.resolve({
            cropRect: { height: 1080, width: 1513, x: 204, y: 0 },
            height: 1000,
            mime: 'image/jpeg',
            modificationDate: '1706166538000',
            path: 'file:///storage/emulated/0/Android/data/id.smiteknologi.shiphire/files/Pictures/553466d2-3a3e-441d-941b-5206a52646fc.jpg',
            size: 303593,
            width: 1400,
        }),
    ),
    types: {
        image: 'image/jpg',
    },
}));

jest.mock('../../../../hooks');
const MainOwnerStack = createNativeStackNavigator<MainOwnerStackParamList>();
const queryClient = new QueryClient();
const mockAdapter = new MockAdapter(httpRequest);

const ownerStackMockComponent = (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainOwnerStack.Navigator>
                    <MainOwnerStack.Screen
                        name="ImageForm"
                        component={ImageForm}
                    />
                    <MainOwnerStack.Screen
                        name="RFQTemplateOwnerManagement"
                        component={RFQTemplateOwnerManagement}
                    />
                </MainOwnerStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

const mockDynamicData = {
    dynamicForms: [
        {
            _id: '65a737c476517457909c4d74',
            dynamicInput: {
                __v: 0,
                _id: '65a737c476517457909c4d9f',
                active: true,
                fieldName: 'shipImage',
                fieldType: 'image',
                formType: 'addShipForm',
                inputType: 'imageSelect',
                label: 'Ship Image',
                order: 1,
                templateType: 'shipImage',
            },
            option: [],
            required: true,
        },
    ],
};

const addShipState = {
    name: 'Ship 123',
    desc: 'Ship Desc',
    category: 'Barge',
    pricePerMonth: '12345',
    length: 1234,
    width: 1234,
    height: 1234,
    facilities: ['facility1'],
    specifications: [
        {
            name: 'Capacity',
            value: '',
        },
    ],
    shipDocument: [
        {
            docExpired: undefined,
            label: 'certificateOfRegister',
            name: 'dummy.pdf',
            uri: 'dummy.pdf',
        },
        {
            docExpired: undefined,
            label: 'seaworthyCertificate',
            name: 'dummy.pdf',
            uri: 'dummy.pdf',
        },
        {
            docExpired: undefined,
            label: 'safetyCertificate',
            name: 'dummy.pdf',
            uri: 'dummy.pdf',
        },
        {
            docExpired: undefined,
            label: 'derattingCertificate',
            name: 'dummy.pdf',
            uri: 'dummy.pdf',
        },
    ],
};

const mockSubmitShip = () => {
    mockAdapter.onPost('/admin/save-ship').reply(200, {
        message: 'success',
    });
    (useSubmitShip as jest.Mock).mockImplementation(() => ({
        mutate: jest.fn().mockImplementation((data, options) => {
            const { onSuccess } = options;
            const mockResponseData = {
                data: {
                    data: mockDynamicData,
                },
            };
            onSuccess(mockResponseData);
        }),
    }));
};

const mockSubmitDocument = () => {
    mockAdapter.onPost('/admin/save-ship-document').reply(200, {
        message: 'success',
    });
    (useSubmitShipDocument as jest.Mock).mockImplementation(() => ({
        mutate: jest.fn().mockImplementation((data, options) => {
            const { onSuccess } = options;
            const mockResponseData = {
                data: {
                    data: mockDynamicData,
                },
            };
            onSuccess(mockResponseData);
        }),
    }));
};
const mockSubmitDocumentError = () => {
    mockAdapter.onPost('/admin/save-ship-document').reply(500, {
        message: 'failed',
    });
    (useSubmitShipDocument as jest.Mock).mockImplementation(() => ({
        mutate: jest.fn().mockImplementation((data, options) => {
            const { onError } = options;
            const mockResponseData = {
                message: 'failed',
            };
            onError(mockResponseData);
        }),
    }));
};

const mockSubmitImage = () => {
    mockAdapter.onPost('/admin/save-ship-image').reply(200, {
        message: 'success',
    });
    (useSubmitShipImage as jest.Mock).mockImplementation(() => ({
        mutate: jest.fn().mockImplementation((data, options) => {
            const { onSuccess } = options;
            const mockResponseData = {
                data: {
                    data: mockDynamicData,
                },
            };
            onSuccess(mockResponseData);
        }),
    }));
};
const mockSubmitImageError = () => {
    mockAdapter.onPost('/admin/save-ship-image').reply(500, {
        message: 'failed',
    });
    (useSubmitShipImage as jest.Mock).mockImplementation(() => ({
        mutate: jest.fn().mockImplementation((data, options) => {
            const { onError } = options;
            const mockResponseData = {
                message: 'error',
            };
            onError(mockResponseData);
        }),
    }));
};

beforeEach(() => {
    (useSelector as jest.Mock).mockReturnValue(addShipState);
    mockAdapter.onGet('/admin/get-dynamic-input').reply(200, {
        message: 'success',
        data: mockDynamicData,
    });
    (useGetAddShipDynamicForm as jest.Mock).mockImplementation(() => ({
        mutate: jest.fn().mockImplementation((data, options) => {
            const { onSuccess } = options;
            const mockResponseData = {
                data: {
                    data: mockDynamicData,
                },
            };
            onSuccess(mockResponseData);
        }),
    }));
});

describe('add ship image screen', () => {
    describe('Snapshot testing', () => {
        it('should render add ship image screen correctly', () => {
            const imageScreen = render(ownerStackMockComponent);
            expect(imageScreen).toMatchSnapshot();
        });
    });
    describe('Unit testing', () => {
        it('should save image, save document and save ship data', async () => {
            mockSubmitShip();
            mockSubmitDocument();
            mockSubmitImage();
            const { getByTestId } = render(ownerStackMockComponent);
            const image = getByTestId('image-shipImage');
            const btnSubmit = getByTestId('dynamic-button');
            fireEvent.press(image);

            fireEvent.press(btnSubmit);

            await waitFor(() => {
                const modal = store.getState().modal;
                expect(modal).toEqual({
                    status: 'success',
                    text: 'Kapal sedang ditambahkan, mohon jangan tinggalkan halaman ini',
                    visible: true,
                });
            });
        });
        it('should handle error when save image error', async () => {
            mockSubmitImageError();
            const { getByTestId } = render(ownerStackMockComponent);
            const image = getByTestId('image-shipImage');
            const btnSubmit = getByTestId('dynamic-button');
            fireEvent.press(image);

            fireEvent.press(btnSubmit);

            await waitFor(() => {
                const modal = store.getState().modal;
                expect(modal).toEqual({
                    status: 'success',
                    text: 'Kapal sedang ditambahkan, mohon jangan tinggalkan halaman ini',
                    visible: true,
                });
            });
        });
        it('should handle error when save document error', async () => {
            mockSubmitDocumentError();
            const { getByTestId } = render(ownerStackMockComponent);
            const image = getByTestId('image-shipImage');
            const btnSubmit = getByTestId('dynamic-button');
            fireEvent.press(image);

            fireEvent.press(btnSubmit);

            await waitFor(() => {
                const modal = store.getState().modal;
                expect(modal).toEqual({
                    status: 'success',
                    text: 'Kapal sedang ditambahkan, mohon jangan tinggalkan halaman ini',
                    visible: true,
                });
            });
        });
    });
});
