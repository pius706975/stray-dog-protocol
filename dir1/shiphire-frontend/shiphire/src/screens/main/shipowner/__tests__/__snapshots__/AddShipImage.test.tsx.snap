// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add ship image screen Snapshot testing should render add ship image screen correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                undefined,
                undefined,
                {
                  "flex": 1,
                },
                undefined,
                {},
                {},
                {},
                {
                  "backgroundColor": "#f0f6fb",
                  "justifyContent": "space-between",
                  "position": "relative",
                },
              ]
            }
            testID="imageFormAddShip"
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "alignItems": undefined,
                  "flex": undefined,
                  "flexGrow": 1,
                  "gap": undefined,
                  "justifyContent": undefined,
                  "marginBottom": undefined,
                  "padding": undefined,
                  "paddingVertical": undefined,
                }
              }
            >
              <View>
                <View
                  marginH-20={true}
                  marginV-26={true}
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {},
                      {
                        "marginHorizontal": 20,
                        "marginVertical": 26,
                      },
                      {},
                      undefined,
                    ]
                  }
                >
                  <View
                    accessibilityLabel="progress bar. 99%"
                    accessible={true}
                    onLayout={[Function]}
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {},
                        {},
                        {},
                        [
                          {
                            "backgroundColor": "#E8ECF0",
                            "borderRadius": 999,
                            "overflow": "hidden",
                          },
                          {
                            "borderRadius": 999,
                            "height": 8,
                          },
                          {
                            "backgroundColor": "#9cd2ff",
                            "height": 10,
                            "marginBottom": 20,
                          },
                        ],
                      ]
                    }
                  />
                  <View
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                        ]
                      }
                    >
                      <View
                        paddingB-5={true}
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "paddingBottom": 5,
                            },
                            {},
                            {},
                            undefined,
                          ]
                        }
                      >
                        <Text
                          numberOfLines={0}
                          style={
                            {
                              "borderBottomWidth": 0,
                              "borderColor": undefined,
                              "color": "#216dab",
                              "fontFamily": "Kanit-Medium",
                              "fontSize": 22,
                              "lineHeight": undefined,
                              "textAlign": undefined,
                            }
                          }
                        >
                          Ship Image
                        </Text>
                      </View>
                      <View
                        paddingB-25={true}
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "paddingBottom": 25,
                            },
                            {},
                            {},
                            undefined,
                          ]
                        }
                      >
                        <View
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": undefined,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessibilityValue={
                            {
                              "max": undefined,
                              "min": undefined,
                              "now": undefined,
                              "text": undefined,
                            }
                          }
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          testID="image-shipImage"
                        >
                          <View
                            centerV={true}
                            paddingB-15={true}
                            row={true}
                            spread={true}
                            style={
                              [
                                undefined,
                                undefined,
                                {
                                  "flex": 1,
                                },
                                undefined,
                                {
                                  "paddingBottom": 15,
                                },
                                {},
                                {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                  "justifyContent": "space-between",
                                },
                                {
                                  "borderBottomColor": "#216dab",
                                  "borderBottomWidth": 1,
                                },
                              ]
                            }
                          >
                            <Text
                              numberOfLines={0}
                              style={
                                {
                                  "borderBottomWidth": 0,
                                  "borderColor": undefined,
                                  "color": "#4d5963",
                                  "fontFamily": "Kanit-Regular",
                                  "fontSize": 18,
                                  "lineHeight": undefined,
                                  "textAlign": undefined,
                                }
                              }
                            >
                              Select file
                            </Text>
                            <RNSVGSvgView
                              align="xMidYMid"
                              bbHeight="24"
                              bbWidth="24"
                              fill="none"
                              focusable={false}
                              height={24}
                              meetOrSlice={0}
                              minX={0}
                              minY={0}
                              style={
                                [
                                  {
                                    "backgroundColor": "transparent",
                                    "borderWidth": 0,
                                  },
                                  {
                                    "flex": 0,
                                    "height": 24,
                                    "width": 24,
                                  },
                                ]
                              }
                              vbHeight={24}
                              vbWidth={24}
                              width={24}
                              xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
    <g clip-path="url(#clip0_77_28470)">
    <path d="M18 13H13V18C13 18.55 12.55 19 12 19C11.45 19 11 18.55 11 18V13H6C5.45 13 5 12.55 5 12C5 11.45 5.45 11 6 11H11V6C11 5.45 11.45 5 12 5C12.55 5 13 5.45 13 6V11H18C18.55 11 19 11.45 19 12C19 12.55 18.55 13 18 13Z" fill="#323232"/>
    </g>
    <defs>
    <clipPath id="clip0_77_28470">
    <rect width="24" height="24" fill="white"/>
    </clipPath>
    </defs>
    </svg>
    
    
    "
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <RNSVGGroup
                                fill={null}
                                propList={
                                  [
                                    "fill",
                                  ]
                                }
                              >
                                <RNSVGGroup
                                  clipPath="clip0_77_28470"
                                  fill={
                                    {
                                      "payload": 4278190080,
                                      "type": 0,
                                    }
                                  }
                                >
                                  <RNSVGPath
                                    d="M18 13H13V18C13 18.55 12.55 19 12 19C11.45 19 11 18.55 11 18V13H6C5.45 13 5 12.55 5 12C5 11.45 5.45 11 6 11H11V6C11 5.45 11.45 5 12 5C12.55 5 13 5.45 13 6V11H18C18.55 11 19 11.45 19 12C19 12.55 18.55 13 18 13Z"
                                    fill={
                                      {
                                        "payload": 4281479730,
                                        "type": 0,
                                      }
                                    }
                                    propList={
                                      [
                                        "fill",
                                      ]
                                    }
                                  />
                                </RNSVGGroup>
                                <RNSVGDefs>
                                  <RNSVGClipPath
                                    fill={
                                      {
                                        "payload": 4278190080,
                                        "type": 0,
                                      }
                                    }
                                    name="clip0_77_28470"
                                  >
                                    <RNSVGRect
                                      fill={
                                        {
                                          "payload": 4294967295,
                                          "type": 0,
                                        }
                                      }
                                      height="24"
                                      propList={
                                        [
                                          "fill",
                                        ]
                                      }
                                      width="24"
                                      x="0"
                                      y="0"
                                    />
                                  </RNSVGClipPath>
                                </RNSVGDefs>
                              </RNSVGGroup>
                            </RNSVGSvgView>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    marginT-16={true}
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {},
                        {
                          "marginTop": 16,
                        },
                        {},
                        undefined,
                      ]
                    }
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#216dab",
                          "borderRadius": 6,
                          "elevation": 2,
                          "flexDirection": "row",
                          "gap": 0,
                          "justifyContent": "center",
                          "paddingHorizontal": 10,
                          "paddingVertical": 6,
                        }
                      }
                      testID="dynamic-button"
                    >
                      <Text
                        numberOfLines={0}
                        style={
                          {
                            "borderBottomWidth": 0,
                            "borderColor": undefined,
                            "color": "#f0f6fb",
                            "fontFamily": "Kanit-Medium",
                            "fontSize": 18,
                            "lineHeight": undefined,
                            "textAlign": undefined,
                          }
                        }
                      >
                        ShipOwner.buttonAddNewShip
                      </Text>
                      <View
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "position": "absolute",
                              "right": 20,
                            },
                          ]
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="ImageForm"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
